src/__test__/components/NotifyByEmail.test.jsx

```diff
@@ -3,96 +3,73 @@ import {
   render, screen,
 } from '@testing-library/react';
 import NotifyByEmail from 'components/NotifyByEmail';
-import configureMockStore from 'redux-mock-store';
-import thunk from 'redux-thunk';
+import { act } from 'react-dom/test-utils';
+
 import { Provider } from 'react-redux';
 import userEvent from '@testing-library/user-event';
 import * as updateExperiment from 'redux/actions/experiments/updateExperiment';
 import initialExperimentsState from 'redux/reducers/experiments/initialState';
 import * as loadProjects from 'redux/actions/projects/loadProjects';
 import * as loadExperiments from 'redux/actions/experiments/loadExperiments';
+import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
+import mockAPI, {
+  generateDefaultMockAPIResponses,
+} from '__test__/test-utils/mockAPI';
+import { makeStore } from 'redux/store';
+import fake from '__test__/test-utils/constants';
+
+let storeState = null;
 
-const mockStore = configureMockStore([thunk]);
-const experimentId = 'someExperiment';
-const activeProjectUuid = 'project1';
-const defaultState = {
-  projects: {
-    meta: {
-      activeProjectUuid,
-    },
-  },
-  experiments: {
-    [experimentId]: {
-      notifyByEmail: false,
-    },
-    ids: [experimentId],
-  },
-};
+const experimentId = fake.EXPERIMENT_ID;
 
 describe('Notify by email component', () => {
   let updateExperimentSpy;
   let loadProjectsSpy;
   let loadExperimentsSpy;
-  let store;
+
   beforeEach(() => {
     jest.clearAllMocks();
     updateExperimentSpy = jest.spyOn(updateExperiment, 'default');
     loadProjectsSpy = jest.spyOn(loadProjects, 'default');
     loadExperimentsSpy = jest.spyOn(loadExperiments, 'default');
+    enableFetchMocks();
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    storeState = makeStore();
   });
-  const renderNotifyByEmail = (state) => {
-    store = mockStore(state);
-    render(
-      <Provider store={store}>
+  const renderNotifyByEmail = async () => {
+    await act(async () => (render(
+      <Provider store={storeState}>
         <NotifyByEmail
           experimentId={experimentId}
         />
       </Provider>,
-    );
+    )));
   };
+  it('loads experiments if  there is an activeProjectUuid', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
+    await renderNotifyByEmail();
+    expect(loadExperimentsSpy).toHaveBeenCalledTimes(1);
+  });
 
   it('Renders Correctly', () => {
-    renderNotifyByEmail(defaultState);
+    renderNotifyByEmail();
```

Comment:
I would `await` here to make sure that `renderNotifyByEmail` is resolved before the next LOC Is executed